Sources/Bookmarks/BookmarkListViewModel.swift

```diff
@@ -138,6 +140,30 @@ public class BookmarkListViewModel: BookmarkListInteracting, ObservableObject {
         refresh()
     }
 
+    private func reattachOrphanedBookmarks(forMoving bookmark: BookmarkEntity, toIndex: Int) {
+        guard let rootFolder = BookmarkUtils.fetchRootFolder(context) else {
+            return
+        }
+
+        let orphanedBookmarks: [BookmarkEntity] = BookmarkUtils.fetchOrphanedEntities(context)
```

Comment:
Do we need to fetch this? Can't we just filter displayed bookmarks to get these that have no parent? I'm afraid if refetching could still lead to index inconsistencies if we have changes happening in the background.

Reply:
Of course we can! Thanks for spotting this, fixing now :)

Reply:
It's fixed and I've also added a pixel to track orphaned bookmarks presence for users not on sync. It's ready for review 🙏